includes/admin/list-tables/class-wc-admin-list-table-products.php

```diff
@@ -125,8 +125,12 @@ public function define_columns( $columns ) {
 		}
 
 		$show_columns['price']        = __( 'Price', 'woocommerce' );
-		$show_columns['product_cat']  = __( 'Categories', 'woocommerce' );
-		$show_columns['product_tag']  = __( 'Tags', 'woocommerce' );
+		if ( wc_product_categories_enabled() ) {
```

Comment:
WordPress already provides a filter to manipulate columns.
https://developer.wordpress.org/reference/hooks/manage_post_type_posts_columns/